streampipes-client-go/streampipes/endpoint.go

```diff
@@ -0,0 +1,72 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package streampipes
+
+import (
+	"errors"
+	"github.com/apache/streampipes/streampipes-client-go/streampipes/config"
+	headers "github.com/apache/streampipes/streampipes-client-go/streampipes/internal/http_headers"
+	"net/http"
+)
+
+type endpoint struct {
+	config config.StreamPipesClientConfig
+}
+
+func (e *endpoint) makeHeaderAndHttpClient(method string, endPointUrl string) (*http.Response, error) {
```

Comment:
The name of the method doesn't reflect the actual method behavior. It actually executes the request and returns the response, rather than making the header and client.